include/deal.II/sundials/ida.h

```diff
@@ -0,0 +1,653 @@
+                           VectorType &solution_dot);
+
+    /**
+     * Clear internal memory and start with clean objects. This function is
+     * called when the simulation start and when the user returns true to a
+     * call to solver_should_restart().
+     *
+     * By default solver_should_restart returns false. If the user needs to
+     * implement, for example, local adaptivity in space, he or she may assign
+     * a different function to solver_should_restart() that performs all mesh
+     * changes, transfers the solution and the solution dot to the new mesh, and
+     * returns true.
+     *
+     * During reset(), both y and yp are checked for consistency, and according to
+     * what was specified as ic_type (if t==initial_time) or reset_type (if
+     * t>initial_time), yp, y, or both are modified to obtain a consistent set
+     * of initial data.
+     *
+     * @param[in] t  The new starting time
+     * @param[in] h  The new (tentative) starting time step
+     * @param[in,out] y   The new (tentative) initial solution
+     * @param[in,out] yp  The new (tentative) initial solution_dot
+     */
+    void reset(const double &t,
+               const double &h,
+               VectorType &y,
+               VectorType &yp);
+
+    /**
+     * Reinit vector to have the right size, MPI communicator, etc.
+     */
+    std::function<void(VectorType &)> reinit_vector;
+
+    /**
+     * Compute residual.
+     *
+     * This function should return:
+     * - 0: Success
+     * - >0: Recoverable error (IDAReinit will be called if this happens, and
+     *       then last function will be attempted again
+     * - <0: Unrecoverable error the computation will be aborted and an assertion
+     *       will be thrown.
+     */
+    std::function<int(const double t,
+                      const VectorType &y,
+                      const VectorType &y_dot,
+                      VectorType &res)> residual;
+
+    /**
+     * Compute Jacobian.
+     *
+     * This function should return:
+     * - 0: Success
+     * - >0: Recoverable error (IDAReinit will be called if this happens, and
+     *       then last function will be attempted again
+     * - <0: Unrecoverable error the computation will be aborted and an assertion
+     *       will be thrown.
+     */
+    std::function<int(const double t,
+                      const VectorType &y,
+                      const VectorType &y_dot,
+                      const double alpha)> setup_jacobian;
+
+    /**
+     * Solve linear system.
+     *
+     * This function should return:
+     * - 0: Success
+     * - >0: Recoverable error (IDAReinit will be called if this happens, and
+     *       then last function will be attempted again
+     * - <0: Unrecoverable error the computation will be aborted and an assertion
+     *       will be thrown.
+     */
+    std::function<int(const VectorType &rhs, VectorType &dst)> solve_jacobian_system;
```

Comment:
can you say what the function arguments mean, and that the function is supposed to compute `dst = J^{-1} rhs`?

Reply:
done. I don't know why it does not show up here.